crates/core/executor/src/events/utils.rs

```diff
@@ -1,22 +1,48 @@
+use serde::Deserialize;
+use serde::Serialize;
 use std::{
     fmt::Display,
     iter::{Map, Peekable},
 };
 
 use rand::{thread_rng, Rng};
+#[derive(Deserialize, Serialize, Debug, Clone, Copy, Default, Eq, Hash, PartialEq)]
```

Comment:
nit: derive should be below the comment.

The comment has a lot of information (good) but it does not fit inline with the standards of the repo.

A better comment would just be

```
A unique identifier for lookups.

We use 4 u32s instead of a u128 to make it compatible with C.
```

Reply:
Fixed